fix: address various style and text issues

HendrikThePendric opened the pull request:
Fixes:
- [DHIS2-13425](https://dhis2.atlassian.net/browse/DHIS2-13425): Renamed the user table column from ""Account disabled?" to "status" and now populating each row with "Active" or "Disabled" text.
- [DHIS2-12910](https://dhis2.atlassian.net/browse/DHIS2-12910): Changed the button texts below the "create user form" when inviting a user by email*.
- [DHIS2-13424](https://dhis2.atlassian.net/browse/DHIS2-13424): Tweaked the texts in the reset password dialog - this dialog can be opened via the context menu in the user table *.
- [DHIS2-14064](https://dhis2.atlassian.net/browse/DHIS2-14064): Gave the bulk user manager - in the create/edit user-group form - some more width, but still kept a max-width with very wide monitors in mind. I also added some extra whitespace to the last column to prevent the content from jumping around too much. There is some extra space there, but possibly in languages that use a lot of characters there might still be a content jump. I didn't want to add too much whitespace, because it could have a negative impact on the readability of the username on narrow screens.

[*] A note for DHIS2-12910 and DHIS2-13424: These issues can only be tested fully by QA if an email server is configured for the system and the user has the required authorities. Developers reviewing this PR are advised to check out this branch locally and running the app. I will add a few instructions:
1. Just change the code below to `emailConfigured={true}` to test DHIS2-12910:
    https://github.com/dhis2/user-app/blob/4fdf8fdf41db5059d55275914154d61acd9fd76f/src/components/UserForm/UserForm.js#L180
1. Just render this unconditionally to see the button in the user-table context-menu that shows the reset-password dialog, to test DHIS2-13424:
https://github.com/dhis2/user-app/blob/4fdf8fdf41db5059d55275914154d61acd9fd76f/src/pages/UserList/ContextMenu/ContextMenu.js#L101-L110

tomzemp reviewed (commented):
Looks good @HendrikThePendric. Seems to fix all the ticket issues.

Regarding the use of ReactFinalForm.FormSpy:

- use useField / useEffect to just update the . I think you might have referenced this on Slack? I tested and was able to put the useField within the InviteUserSection and it seems to work: https://github.com/dhis2/user-app/commit/922d225385965d4110c939a1c38eca9264a4160a). I don't know if that's much of an improvement (e.g. I don't know how much lighter useField is vs. FormSpy, but I guess it would make the logic slightly clearer?
- Alternatively, maybe it would be easier to just move the InviteUserSection component outside of the form itself (and keep track of whether it's an invite or not in state, which then gets passed to form/handleSubmit, etc.,)? It seems conceptually, like this question is outside of the form itself and that you're generating two different forms based on whether it's an invite or not (hence you get different submit/cancel text for two forms)

Also, not related to this PR directly, but I noticed when hacking to get the 'reset password' option, but I'm wondering if the check of the authorities for this https://github.com/dhis2/user-app/blob/master/src/pages/UserList/ContextMenu/ContextMenu.js#L58-L59 should also account for users that have the authority 'ALL'?

HendrikThePendric commented:
Thanks @tomzemp I liked your suggested changed and have incorporated them into this PR (did something wrong when resolving some merge conflicts which I corrected in [9649afe](https://github.com/dhis2/user-app/pull/1091/commits/9649afe1eeccf7454da548abc005dab2682c5a25)). Also a good point about the `ALL` authority. I also noticed this was wrong while working but forgot to address it in the end. So I have added a commit that fixes this too, in [8ce07d7](https://github.com/dhis2/user-app/pull/1091/commits/8ce07d7b278a95693c619cd61922ee74d42def3b).

tomzemp reviewed (approved):
looks good!
